packages/application-shell-connectors/src/hooks/custom-views/use-custom-view-integration.ts

```diff
@@ -0,0 +1,30 @@
+import { useCallback, useEffect } from 'react';
+
+export type TCustomViewIntegrationParams = {
+  locators: string[];
+  onRefreshDataRequested: (context: Record<string, unknown>) => void;
+};
+
+export const useCustomViewIntegration = (
+  params: TCustomViewIntegrationParams
+) => {
+  const onCustomViewEventHandler = useCallback(
+    (event: CustomEvent<{ originLocatorCode: string }>) => {
+      const { originLocatorCode } = event.detail;
+      if (params.locators.includes(originLocatorCode)) {
+        params.onRefreshDataRequested(event.detail);
+      }
+    },
+    []
+  ) as EventListener;
+
+  useEffect(() => {
+    window.addEventListener('custom-view-closed', onCustomViewEventHandler);
```

Comment:
Nit: define the events in the `CUSTOM_VIEWS_EVENTS_NAMES` in the `constants` package

https://github.com/commercetools/merchant-center-application-kit/blob/e02bccd6aaac8e055a5f4c1376aafbfbcc12bb34/packages/constants/src/constants.ts#L335-L341